packages/purpose-process/src/services/purposeService.ts

```diff
@@ -1186,6 +1185,8 @@ export function purposeServiceBuilder(
 
       const purposeToClone = await retrievePurpose(purposeId, readModelService);
 
+      assertRequesterCanActAsConsumer(purposeToClone.data, authData, undefined);
```

Comment:
In this way we allow the original consumer to clone any purpose, delegated or not, which is not what we want. 
This assertion always passes when the requester is the original consumer, without any check on any delegation -- instead of `undefined` we should retrieve and pass the delegation related to the purpose, if present, so that:
1. the original consumer can clone only their purposes with no delegation
2. the delegate (if present) can clone only delegated purposes under that delegation.

See also:
* this message as reference: https://buildo.slack.com/archives/C07U5M9SJ1Y/p1736934269029509?thread_ts=1736932336.366719&cid=C07U5M9SJ1Y
* #1368 that did the same for other purpose calls

```suggestion
      assertRequesterCanActAsConsumer(
        purpose.data,
        authData,
        purpose.data.delegationId &&
          (await readModelService.getActiveConsumerDelegationByDelegationId(
            purpose.data.delegationId
          ))
      );
```